src/panels/lovelace/cards/hui-weather-forecast-card.ts

```diff
@@ -0,0 +1,458 @@
+                        "ui.card.weather.attributes.wind_speed"
+                      )}:
+                      <span class="measurand">
+                        ${this.getWindSpeed(stateObj.attributes.wind_speed)}
+                      </span>
+                      ${this.getWindBearing(stateObj.attributes.wind_bearing)}
+                    </div>
+                  `
+                : ""}
+            </div>
+          </div>
+          ${forecast
+            ? html`
+                <div class="forecast">
+                  ${forecast.map(
+                    (item) => html`
+                      <div>
+                        <div class="weekday">
+                          ${this.computeDate(item.datetime)}<br />
+                          ${!this._showValue(item.templow)
+                            ? html`
+                                ${this.computeTime(item.datetime)}
+                              `
+                            : ""}
+                        </div>
+                        ${this._showValue(item.condition)
+                          ? html`
+                              <div class="icon">
+                                <ha-icon
+                                  .icon="${this.getWeatherIcon(item.condition)}"
+                                ></ha-icon>
+                              </div>
+                            `
+                          : ""}
+                        ${this._showValue(item.temperature)
+                          ? html`
+                              <div class="temp">
+                                ${item.temperature}
+                                ${this.getUnit("temperature")}
+                              </div>
+                            `
+                          : ""}
+                        ${this._showValue(item.templow)
+                          ? html`
+                              <div class="templow">
+                                ${item.templow} ${this.getUnit("temperature")}
+                              </div>
+                            `
+                          : ""}
+                        ${this._showValue(item.precipitation)
+                          ? html`
+                              <div class="precipitation">
+                                ${item.precipitation}
+                                ${this.getUnit("precipitation")}
+                              </div>
+                            `
+                          : ""}
+                      </div>
+                    `
+                  )}
+                </div>
+              `
+            : ""}
+        </div>
+      </ha-card>
+    `;
+  }
+
+  protected shouldUpdate(changedProps: PropertyValues): boolean {
+    return hasConfigOrEntityChanged(this, changedProps);
+  }
+
+  private handleClick(): void {
+    fireEvent(this, "hass-more-info", { entityId: this._config!.entity });
+  }
+
+  private computeForecast(forecast: any[]): any[] {
```

Comment:
`any` ?

Reply:
I also think that we should just inline this. Could be computed inside the render function before you return the template if you need to use it twice.

Reply:
I'm lazy :rofl: